src/core/components/LoginPage/index.js

```diff
@@ -20,7 +21,7 @@ export default class LoginPage extends React.Component {
           {message || _('You must be logged in to access this page.')}
         </p>
         <p>
-          <a className="button" href={startLoginUrl()}>
+          <a className="button" href={startLoginUrl({ location })}>
```

Comment:
I'm not sure what there really is to fix up. There could be a login button component but it would still need to be passed the location. So I think you'd either import a component that knows about `startLoginUrl` or import `startLoginUrl` directly. Unless we have a specific button we always want to use I'd say we just leave it like this.